src/snowflake/snowpark/mock/select_statement.py

```diff
@@ -435,6 +435,10 @@ def set_operator(
         return new
 
     def limit(self, n: int, *, offset: int = 0) -> "SelectStatement":
+        if offset != 0:
+            raise NotImplementedError(
+                "[Local Testing] Non-zero offset is not currently supported."
+            )
```

Comment:
what's the case of this one?

I tried the following code looks like `limit` is working:

```python
    origin_df: DataFrame = session.create_dataframe(
        [
            [1, "a", True],
            [6, "c", False],
            [None, None, None],
        ],
        schema=["m", "n", "o"],
    )
    assert origin_df.select("m", "n", "o").limit(2).collect() == [
        Row(1, "a", True),
        Row(6, "c", False)
    ]
```